### ISSUE TITLE
StringUtils test partially fail under MSVC/Windows

### ISSUE THREAD

**Issue by Framstag** [Framstag/libosmscout#1082]
See https://github.com/Framstag/libosmscout/runs/2535134654?check_suite_focus=true

It seems that some tests fails.

See also depreciated warning during build...

The appveyor MSVC/meson build does not execute the tests so this problem was up to now undetected.

**Comment by janbar** [comment]
Hi @Framstag ,

The Microsoft compiler assumes source is ANSI-encoded, which depends on the localized version of Windows in use. On U.S. and Western European Windows the encoding is assumed to be is Windows-1252.
I tested removing u8 modifiers and it works. So we need to remove all u8 modifiers in all source files to avoid this behavior with the Microsoft compiler.
Else I refactored the utf8helper using C++ class without too much overhead. It allows to inject customized functor if @Karry want to adapt i.e the normilze functor as he need.
I will make 2 PR , one to fix de MS issue and the other for the refactoring.
